content/pages/contributing.md

```diff
@@ -359,6 +362,26 @@ $ git commit
 
 Which will prompt you for a log message. Please provide something informative. If you make lots of changes, it is best to make multiple commits, broken up into related chunks. E.g. “fixed x”, “added documentation”, “added testing”.
 
+Depending on which repo you are making changes to, a `pre-commit`
+[hook](https://www.atlassian.com/git/tutorials/git-hooks)
+may be enabled that will validate your changes and even make changes
+on its own (such as cleaning up code formatting). If the _hook_ makes
+any changes you may need to run `git commit` a second time to commit the
```

Comment:
This should specify that you need to do
``` 
git add PATH_TO_NEW_FILE
```
one more time to add the new changes to your commit, and only then do another `git commit`.

Reply:
Hmm. I didn't think 'git add' was needed unless you added a new file to the repo. I've only had to run 'git commit'. Is there a possibility that the hook will add a new file, or is my understanding incorrect?

Reply:
In my experience, the hook will make changes, but you still need to `git add` those changes for them to be included in the commit.

It may be that there are some configuration issues with `precommit` that change this behavior. @andersy005 ?

Reply:
It's not creating a new file, it's making modifications to an existing file. git will then treat these modifications just like any other. Until you explicitly `add` them to the staging area, those changes are not part of the commit.

Reply:
Uhh. Not trying to be difficult, but this doesn't jive with my understanding (and use of git). My understanding/usage:

1. Use 'git add' the first time you add a file to your local .git repo. This step need never be repeated again (I think)
2. Thereafter use 'git commit' whenever changes are made to a file that you want to update the local repo with.

Do you use different (preferred) process? Just trying to understand

Reply:
Ok, I think I see the disconnect.

Any git commit is a two-step process. Files are added to the "staging area" with `git add FILENAME`, and anything in the staging area is committed with `git commit`.

It's possible to fast-forward through the two-step process by doing
```
git commit FILENAME
```
which is equivalent to 
```
git add FILENAME
git commit
```

But if you make some changes to a previously committed file, and then just do 
```
git commit
```
(without specifying which files to commit), then nothing will happen.

Reply:
So in the instructions where you say "you may need to run `git commit` a second time",  maybe better to say "run `git commit FILENAME` a second time", or spell out the two-stage process.

Does that make more sense?

Reply:
Usually @dopplershift is better at explaining this kind of stuff...

Reply:
I personally use both commands `git add` & `git commit` regardless of whether it's a new file or an existing one... :)

Reply:
Now I see the disconnect too. Basically I forgot the file name.  Please see latest revisions.

Reply:
> So in the instructions where you say "you may need to run `git commit` a second time", maybe better to say "run `git commit FILENAME` a second time", or spell out the two-stage process.
> 
> Does that make more sense?

It does make more sense. I think that is what I have in there. Maybe we crossed wires. Could you please refresh and look?

Reply:
> I personally use both commands `git add` & `git commit` regardless of whether it's a new file or an existing one... :)

Is there a preference for one over the other, or is it fine to leave it as is?

Reply:
Ok, looks good to me now! I think the instructions are clear and the new callout note helps make things clear. 

This has been interesting. I guess I learned one workflow year ago (git add, git commit), and I've never revisited it until now.